src/database/engine/volatile_vector/mod.rs

```diff
@@ -0,0 +1,70 @@
+//! # Volatile Database
```

Comment:
Move this file out of `engine`, and delete the `engine` directory. The file tree should look like this:

```
src/
  database/
    volatile/
      ...
    vector/
      ...
```

on that note, the `vector` database is not supposed to be volatile. I still added comments below just in case you were wondering why stuff wasn't working, but the `vector` database implementation should be persistent (and tabular, and so on)